docs/proposals/initial-resources.md

```diff
@@ -0,0 +1,57 @@
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest 1.0.x release of this document can be found
+[here](http://releases.k8s.io/release-1.0/docs/proposals/initial-resources.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+Since we want to make Kubernetes as simple as possible for its users we don’t want to require setting [Resources](../../pkg/api/v1/types.go#L696) for container by its owner. On the other hand having Resources filled is critical for scheduling decisions. Current solution to set up Resources to hardcoded value has obvious drawbacks. We need to implement a component which will set initial Resources to a reasonable value.
+
+### Design
+
+The component will be implemented as an [admission plugin](../../plugin/pkg/admission/) and invoked right before [LimitRanger](../../cluster/gce/config-default.sh#L91). For every container without Resources specified it will try to predict amount of resources that should be sufficient for it. So that a pod without specified resources will be treated as Burstable (see more details [#11713](https://github.com/GoogleCloudPlatform/kubernetes/pull/11713)). The component will set only [Request](https://github.com/GoogleCloudPlatform/kubernetes/blob/3d2d99c6fd920386eea4ec050164839ec6db38f0/pkg/api/types.go#L691) field in the first version to avoid killing containers due to OOM (however the container still may be preempted if exceeds requested resources). To make the component work with LimitRanger the estimated value will be capped by min and max possible values if defined. It will prevent from situation when the pod is rejected due to too low or too high estimation. The container won’t be marked as managed by this component in any way, however appropriate event will be exported. The predicting algorithm should be simple to not increase significantly e2e pod startup latency [#3954](https://github.com/GoogleCloudPlatform/kubernetes/pull/3954).
+
+### Algorithm
+
+In the first version estimation will be made based on historical data for the Docker image being run in the container (both the name and the tag matters). CPU/memory usage of each container is exported periodically (by default with 1 minute resolution) to the backend. The component will set Request for both cpu/mem as the 90th percentile of such samples (for exactly the same image - the same name and tag) from last 30 days and based on no more than 1000 youngest samples. If there is not enough data (at least 10 samples) it will take also samples with the same name and possible different tags. If there is still no data the default value will be used. All those values will be easily configurable with appropriate flags.
```

Comment:
>(...) would it be better to leave the >empty in this case and let LimitRanger (or other admission control step) set the default?

That's how it will be implemented. Added info in text.

> Oh, I see you cover that later in a separate section ("Monitoring pipeline"). You should make a forward reference to that section. Also consider adding some detail to that section.

Added forward reference. What kind of details do you expect?

>I can understand using 90th %ile for CPU but I wonder if you should use a higher percentile for memory?

The algorithm should be definitely tuned on some real data. 90th seems to be the be a good guess for the first version. We don't want to set this too high to not block scheduling stuff on the cluster.